src/applications/renderer/src/components/renderer/component.js

```diff
@@ -51,11 +67,13 @@ const Renderer = ({
     && chartWidgetConfig?.legend?.length > 0
     && !thumbnail;
 
+  const hasNoData = dataURL && !widgetData || widgetData?.length === 0;
```

Comment:
Could you put parentheses? I guess this is what you meant:
```suggestion
  const hasNoData = dataURL && (!widgetData || widgetData?.length === 0);
```